### PR TITLE
add in-game auto announcement

### PR DESCRIPTION
**Pull request by ImYenil**
add arena announcement

### CODE REVIEW

**Comment by tomkeuper** [comment]
Add an option to disable announcements? a boolean enabled will be fine

**Comment by ImYenil** [comment]
> Add an option to disable announcements? a boolean enabled will be fine

will add

**Comment by Xxyuri2005xX** [comment]
can you also send a video that shows how it works?

File: bedwars-plugin/src/main/java/com/tomkeuper/bedwars/configuration/MainConfig.java [hunk 1]
```diff
@@ -78,6 +78,7 @@ public MainConfig(Plugin plugin, String name) {
         yml.addDefault(ConfigPath.GENERAL_CONFIGURATION_LEAVE_DELAY, 3);
         yml.addDefault(ConfigPath.GENERAL_CONFIGURATION_REJOIN_TIME, 60 * 5);
         yml.addDefault(ConfigPath.GENERAL_CONFIGURATION_RE_SPAWN_INVULNERABILITY, 4000);
+        yml.addDefault(ConfigPath.GENERAL_CONFIGURATION_ANNOUNCEMENT_SEND_EVERY, 6000);
```
Comment (Xxyuri2005xX) [inline bedwars-plugin/src/main/java/com/tomkeuper/bedwars/configuration/MainConfig.java]: how much time is this 6000? every minute? 30 seconds? 2 minutes?

Comment (ImYenil) [reply bedwars-plugin/src/main/java/com/tomkeuper/bedwars/configuration/MainConfig.java]: > how much time is this 6000? every minute? 30 seconds? 2 minutes?

5 minutes

**Comment by ImYenil** [comment]
> can you also send a video that shows how it works?


https://github.com/tomkeuper/BedWars2023/assets/85287520/d9b078a0-c043-4f6f-ae52-d057cdf48989

File: bedwars-plugin/src/main/java/com/tomkeuper/bedwars/configuration/MainConfig.java (hunk 1 above)
Comment (Xxyuri2005xX) [reply bedwars-plugin/src/main/java/com/tomkeuper/bedwars/configuration/MainConfig.java]: maybe set this to 300 and add in the taskTimer a `*20`

Comment (ImYenil) [reply bedwars-plugin/src/main/java/com/tomkeuper/bedwars/configuration/MainConfig.java]: > maybe set this to 300 and add in the taskTimer a `*20`

will change

**Review by porudev — APPROVED** [review]

**Comment by Xxyuri2005xX** [comment]
> > can you also send a video that shows how it works?
> 
>  Base.Profile.2023.05.21.-.22.41.16.01.mp4

ok thanks for the video

File: bedwars-plugin/src/main/java/com/tomkeuper/bedwars/configuration/MainConfig.java (hunk 1 above)
Comment (Xxyuri2005xX) [reply bedwars-plugin/src/main/java/com/tomkeuper/bedwars/configuration/MainConfig.java]: ok thanks

File: bedwars-api/src/main/java/com/tomkeuper/bedwars/api/configuration/ConfigPath.java [hunk 2]
```diff
@@ -121,6 +121,7 @@ public class ConfigPath {
     public static final String GENERAL_CONFIGURATION_ARENA_GROUPS = "arenaGroups";
     public static final String GENERAL_CONFIGURATION_REJOIN_TIME = "rejoin-time";
     public static final String GENERAL_CONFIGURATION_RE_SPAWN_INVULNERABILITY = "re-spawn-invulnerability";
+    public static final String GENERAL_CONFIGURATION_ANNOUNCEMENT_SEND_EVERY = "announcement-send-every";
```
Comment (Xxyuri2005xX) [inline bedwars-api/src/main/java/com/tomkeuper/bedwars/api/configuration/ConfigPath.java]: idk here but maybe use that: `in-game-announcement-cooldown` and `GENERAL_CONFIGURATION_IN_GAME_ANNOUNCEMENT_COOLDOW`

Comment (ImYenil) [reply bedwars-api/src/main/java/com/tomkeuper/bedwars/api/configuration/ConfigPath.java]: > idk here but maybe use that: `in-game-announcement-cooldown` and `GENERAL_CONFIGURATION_IN_GAME_ANNOUNCEMENT_COOLDOW`

Okay. I will change everything

Comment (ImYenil) [reply bedwars-api/src/main/java/com/tomkeuper/bedwars/api/configuration/ConfigPath.java]: > > idk here but maybe use that: `in-game-announcement-cooldown` and `GENERAL_CONFIGURATION_IN_GAME_ANNOUNCEMENT_COOLDOW`
> 
> Okay. I will change everything

done changing everything

File: bedwars-api/src/main/java/com/tomkeuper/bedwars/api/language/Messages.java [hunk 3]
```diff
@@ -152,6 +152,7 @@ public class Messages {
 
     public static String ARENA_MAX_BUILD_LIMIT_REACHED = "build-max-height-limit-reached";
     public static String ARENA_MIN_BUILD_LIMIT_REACHED = "build-min-height-limit-reached";
+    public static String ARENA_ANNOUNCEMENT = "arena-announcement";
```
Comment (Xxyuri2005xX) [inline bedwars-api/src/main/java/com/tomkeuper/bedwars/api/language/Messages.java]: if you can change `ARENA_ANNOUNCEMENT` and `arena-announcement` to `IN_GAME_ANNOUNCEMENT` and `in-game-announcement` it's perfect.I noticed that only now.

Comment (tomkeuper) [reply bedwars-api/src/main/java/com/tomkeuper/bedwars/api/language/Messages.java]: maybe ARENA_IN_GAME_ANNOUNCEMENT is better as its an ARENA setting

Comment (Xxyuri2005xX) [reply bedwars-api/src/main/java/com/tomkeuper/bedwars/api/language/Messages.java]: also

File: bedwars-plugin/src/main/java/com/tomkeuper/bedwars/arena/Announcement.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+                    return;
+                }
+                for (Player player : arena.getPlayers()) {
+                    if (arena.getStatus() == GameState.playing) {
+                        player.sendMessage(announcement);
+                    }
+                }
+                ++this.index;
+            }
+        }.runTaskTimer(BedWars.plugin, config.getInt(ConfigPath.GENERAL_CONFIGURATION_IN_GAME_ANNOUNCEMENT_COOLDOW) * 20L, config.getInt(ConfigPath.GENERAL_CONFIGURATION_IN_GAME_ANNOUNCEMENT_COOLDOW) *20L);
```
Comment (tomkeuper) [inline bedwars-plugin/src/main/java/com/tomkeuper/bedwars/arena/Announcement.java]: Maybe you can run this async with runTaskTimerAsynchronously

Comment (ImYenil) [reply bedwars-plugin/src/main/java/com/tomkeuper/bedwars/arena/Announcement.java]: > Maybe you can run this async with runTaskTimerAsynchronously

done

**Comment by Xxyuri2005xX** [comment]
perfect

**Review by Xxyuri2005xX — APPROVED** [review]

**Pull request merged**